src/services/AzureResourcesService.ts

```diff
@@ -1,27 +1,30 @@
-import type { GenericResource, ResourceGroup, ResourceManagementClient } from "@azure/arm-resources";
+import { GenericResource, ResourceGroup, ResourceManagementClient } from "@azure/arm-resources";
+import { getSessionFromVSCode } from "@microsoft/vscode-azext-azureauth";
 import { uiUtils } from "@microsoft/vscode-azext-azureutils";
-import { createSubscriptionContext, IActionContext } from "@microsoft/vscode-azext-utils";
+import { createCredential } from "@microsoft/vscode-azext-utils";
 import { AzureSubscription } from "api/src/resources/azure";
 import { ext } from "../extensionVariables";
-import { createResourceClient } from "../utils/azureClients";
 
 export interface AzureResourcesService {
-    listResources(context: IActionContext, subscription: AzureSubscription): Promise<GenericResource[]>;
-    listResourceGroups(context: IActionContext, subscription: AzureSubscription): Promise<ResourceGroup[]>;
+    listResources(subscription: AzureSubscription): Promise<GenericResource[]>;
+    listResourceGroups(subscription: AzureSubscription): Promise<ResourceGroup[]>;
 }
 
 export const defaultAzureResourcesServiceFactory = (): AzureResourcesService => {
-    async function createClient(context: IActionContext, subscription: AzureSubscription): Promise<ResourceManagementClient> {
-        const subContext = createSubscriptionContext(subscription);
-        return await createResourceClient([context, subContext]);
+    async function createClient(subscription: AzureSubscription): Promise<ResourceManagementClient> {
+        const session = await getSessionFromVSCode("https://management.azure.com//.default", subscription.tenantId, { createIfNone: false, silent: true, account: subscription.account })
+        const credential = createCredential(() => session);
+        const client = new ResourceManagementClient(credential, subscription.subscriptionId);
```

Comment:
We need to still be using `createResourceClient`. That adds a ton of extra stuff including support for the MFA challenges that we need.